### ISSUE TITLE
sendMessage::error Error: Evaluation failed: TypeError: Cannot read property 'WidFactory' of undefined

### ISSUE THREAD

**Issue by glbatistabh** [pedroslopez/whatsapp-web.js#330]
Is anyone else having this problem when sending a message with image attached?

```
sendMessage::error Error: Evaluation failed: TypeError: Cannot read property 'WidFactory' of undefined
    at __puppeteer_evaluation_script__:2:42
    at ExecutionContext._evaluateInternal (/home/admin/whats-api-test/node_modules/puppeteer/lib/ExecutionContext.js:102:19)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  -- ASYNC --
    at ExecutionContext.<anonymous> (/home/admin/whats-api-test/node_modules/puppeteer/lib/helper.js:94:19)
    at DOMWorld.evaluate (/home/admin/whats-api-test/node_modules/puppeteer/lib/DOMWorld.js:89:24)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  -- ASYNC --
    at Frame.<anonymous> (/home/admin/whats-api-test/node_modules/puppeteer/lib/helper.js:94:19)
    at Page.evaluate (/home/admin/whats-api-test/node_modules/puppeteer/lib/Page.js:612:14)
    at Page.<anonymous> (/home/admin/whats-api-test/node_modules/puppeteer/lib/helper.js:95:27)
    at Client.sendMessage (/home/admin/whats-api-test/node_modules/whatsapp-web.js/src/Client.js:461:47)
    at /home/admin/whats-api-test/core/server/api/whatsapp/send-message.js:81:24
    at nextTask (/home/admin/whats-api-test/node_modules/async/dist/async.js:4576:27)
    at next (/home/admin/whats-api-test/node_modules/async/dist/async.js:4584:13)
    at /home/admin/whats-api-test/node_modules/async/dist/async.js:321:20
    at /home/admin/whats-api-test/core/server/api/whatsapp/send-message.js:64:14
    at nextTask (/home/admin/whats-api-test/node_modules/async/dist/async.js:4576:27)
  -- ASYNC --
    at Page.<anonymous> (/home/admin/whats-api-test/node_modules/puppeteer/lib/helper.js:94:19)
    at Client.sendMessage (/home/admin/whats-api-test/node_modules/whatsapp-web.js/src/Client.js:461:47)
    at /home/admin/whats-api-test/core/server/api/whatsapp/send-message.js:81:24
    at nextTask (/home/admin/whats-api-test/node_modules/async/dist/async.js:4576:27)
    at next (/home/admin/whats-api-test/node_modules/async/dist/async.js:4584:13)
    at /home/admin/whats-api-test/node_modules/async/dist/async.js:321:20
    at /home/admin/whats-api-test/core/server/api/whatsapp/send-message.js:64:14
    at nextTask (/home/admin/whats-api-test/node_modules/async/dist/async.js:4576:27)
    at next (/home/admin/whats-api-test/node_modules/async/dist/async.js:4584:13)
    at /home/admin/whats-api-test/node_modules/async/dist/async.js:321:20
```

### Environment (please complete the following information):
**WhatsApp**
 - Account type: Standard
 - Device OS: Android

**Library**
 - Browser: Chromium (Chromium r782078 / automatic download by puppeteer)
 - NodeJS version: v10.22.0
 - npm or yarn version: 6.14.7
 - whatsapp-web.js version: last (whatsapp-web.js@1.8.1-post)
 - SO: Linux ip-172-26-3-25 4.9.0-13-amd64 #1 SMP Debian 4.9.228-1 (2020-07-05) x86_64 GNU/Linux

**Comment by pedroslopez** [comment]
It's still working fine for me. Please ensure the client has been properly initialized- this error seems like it hasn't.

Ensure you're calling these functions once the client is ready (`ready` event has been emitted) if you're not doing it as a response to a message.

**Comment by glbatistabh** [comment]
> It's still working fine for me. Please ensure the client has been properly initialized- this error seems like it hasn't.
> 
> Ensure you're calling these functions once the client is ready (`ready` event has been emitted) if you're not doing it as a response to a message.

Thanks for the feedback. I'll check.

**Comment by lulukamagaiver** [comment]
When using the lib, I am having this error when sending the message.

> Error: Evaluation failed: TypeError: Cannot read property 'WidFactory' of undefined
>     at __puppeteer_evaluation_script__:2:42
version 1.23.0

**Comment by RDRT1988** [comment]
Buen dia panitas, yo estoy presentando el problema hoy, reinstale la libreria y continuo igual

Error fn sendMessageAgent ❌
 Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'WidFactory')
    at __puppeteer_evaluation_script__:2:42
    at ExecutionContext._evaluateInternal (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/1tb/NodeJS/COS_RPA_BOTWHATSAPPTERPELIVR/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/1tb/NodeJS/COS_RPA_BOTWHATSAPPTERPELIVR/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
    at async Client.sendMessage (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/1tb/NodeJS/COS_RPA_BOTWHATSAPPTERPELIVR/node_modules/whatsapp-web.js/src/Client.js:888:28)
    at async /mnt/8047c310-52a3-4474-a822-5e94f93beb94/1tb/NodeJS/COS_RPA_BOTWHATSAPPTERPELIVR/apis/bot-whatsapp-webjs/index.js:657:19
❌ Error en ruta POST /sendMessage {
  error: true,
  msg: Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'WidFactory')
      at __puppeteer_evaluation_script__:2:42
      at ExecutionContext._evaluateInternal (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/1tb/NodeJS/COS_RPA_BOTWHATSAPPTERPELIVR/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async ExecutionContext.evaluate (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/1tb/NodeJS/COS_RPA_BOTWHATSAPPTERPELIVR/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
      at async Client.sendMessage (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/1tb/NodeJS/COS_RPA_BOTWHATSAPPTERPELIVR/node_modules/whatsapp-web.js/src/Client.js:888:28)
      at async /mnt/8047c310-52a3-4474-a822-5e94f93beb94/1tb/NodeJS/COS_RPA_BOTWHATSAPPTERPELIVR/apis/bot-whatsapp-webjs/index.js:657:19
}

**Comment by RDRT1988** [comment]
Buen dia panitas, yo estoy presentando el problema hoy, reinstale la libreria y continuo igual

Error fn sendMessageAgent ❌
 Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'WidFactory')
    at __puppeteer_evaluation_script__:2:42
    at ExecutionContext._evaluateInternal (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/proyecto/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async ExecutionContext.evaluate (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/proyecto/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
    at async Client.sendMessage (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/proyecto/node_modules/whatsapp-web.js/src/Client.js:888:28)
    at async /mnt/8047c310-52a3-4474-a822-5e94f93beb94/proyecto/apis/bot-whatsapp-webjs/index.js:657:19
❌ Error en ruta POST /sendMessage {
  error: true,
  msg: Error: Evaluation failed: TypeError: Cannot read properties of undefined (reading 'WidFactory')
      at __puppeteer_evaluation_script__:2:42
      at ExecutionContext._evaluateInternal (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/proyecto/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:221:19)
      at processTicksAndRejections (node:internal/process/task_queues:96:5)
      at async ExecutionContext.evaluate (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/proyecto/node_modules/puppeteer/lib/cjs/puppeteer/common/ExecutionContext.js:110:16)
      at async Client.sendMessage (/mnt/8047c310-52a3-4474-a822-5e94f93beb94/proyecto/node_modules/whatsapp-web.js/src/Client.js:888:28)
      at async /mnt/8047c310-52a3-4474-a822-5e94f93beb94/proyecto/apis/bot-whatsapp-webjs/index.js:657:19
}

Quedo atento...

**Comment by smsoousa** [comment]
I get the same problem in version 1.23.0. Any solution? See that they are closing conversations without solving the problem.

**Comment by lulukamagaiver** [comment]
That. I've already done an update, but the problem still persists.

**Comment by guiac** [comment]
any solution?

**Comment by group2tts** [comment]
any solution?

**Comment by SouryuuAsuka** [comment]
any solution?)
